### PR TITLE
Feature/440 - Remove edit lock after mutation is done

### LINKED ISSUE #440: Should be able to remove post edit lock
We currently have a need to remove the `edit_lock` that gets added after a post is inserted/updated. We think it's causing some issues where we are trying to update a post while it's locked, and it creates a duplicate post.

To solve this, I'd like to add a new hook like `graphql_post_object_mutation_before_update_additional_data` where the `set_edit_lock` method would get added to. That way people could remove the callback externally.

### When reporting a _bug_, please be sure to include the following:
- [ ] A descriptive title
- [ ] An *isolated* way to reproduce the behavior (example: GitHub repository with code isolated to the issue that anyone can clone to observe the problem)
- [ ] What version of `WPGraphQL` you're using, and the platform(s) you're running it on
- [ ] What other plugins you're using
- [ ] The behavior you expect to see, and the actual behavior

### When you open an _issue_ for a feature request, please add as much detail as possible:
- [x] A descriptive title
- [x] A description of the problem you're trying to solve, including *why* you think this is a problem
- [x] An overview of the suggested solution
- [x] If the feature changes current behavior, reasons why your solution is better

### PR DESCRIPTION
**Pull request by CodeProKid**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
…

Does this close any currently open issues?
------------------------------------------
Closes #440 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
Looks like master needs to get pulled through develop, eh?

LMK if this looks good or not, I will write some tests for this as well.

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

File: src/Type/PostObject/Mutation/PostObjectMutation.php [hunk 1]
```diff
@@ -287,9 +287,25 @@ public static function prepare_post_object( $input, $post_type_object, $mutation
...
+		 * @param string        $mutation_name        The name of the mutation (ex: create, update, delete)
+		 * @param AppContext    $context              The AppContext passed down to all resolvers
+		 * @param ResolveInfo   $info                 The ResolveInfo passed down to all resolvers
+		 * @param string        $intended_post_status The intended post_status the post should have according to the mutation input
+		 * @param string        $default_post_status  The default status posts should use if an intended status wasn't set
+		 *
+		 * @return bool
 		 */
-		self::set_edit_lock( $post_id );
+		if ( true === apply_filters( 'graphql_lock_post_after_mutation', true, $post_id, $input, $post_type_object, $mutation_name, $context, $info, $default_post_status, $intended_post_status ) ) {
```
[the remark this answers is not in the document]
Comment (jasonbahl) [reply src/Type/PostObject/Mutation/PostObjectMutation.php]: or, if you're using the same filter for both setting the lock and removing it, maybe just `graphql_post_object_mutation_set_edit_lock`? 🤔

[the remark this answers is not in the document]
Comment (CodeProKid) [reply src/Type/PostObject/Mutation/PostObjectMutation.php]: yeah, that second one makes more sense 👍

**Comment by CodeProKid** [comment]
@jasonbahl adjusted filter name in most recent commit.

**Comment by CodeProKid** [comment]
@jasonbahl tests passed. LMK if this is good to go 👍

**Comment by jasonbahl** [comment]
@CodeProKid looks good. Thanks!

**Pull request merged**